src/components/browserSupport/hooks/useBrowserSupport.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+
+// libraries
+import React, { useCallback, useEffect, useState } from 'react'
+import { noop } from 'lodash'
+
+// components
+import UnsupportedDialog from '../components/UnsupportedDialog'
+import SwitchToSafariDialog from '../components/SwitchToSafariDialog'
+
+// hooks
+import useMountedState from '../../../lib/hooks/useMountedState'
+import { useDialog } from '../../../lib/undux/utils/dialog'
+
+// utils
+import logger from '../../../lib/logger/pino-logger'
+import { isBrowser, isChrome, isIOSWeb, isMobileWeb, isSafari } from '../../../lib/utils/platform'
+
+const log = logger.child({ from: 'useBrowserSupport' })
+
+export default (options = {}) => {
+  const {
+    onlyIOS = false,
```

Comment:
why this is needed ? for IOS we just showing another dialog.

Reply:
why onCheck was removed? it allows you to specify custom compatibility checker if you want to have some custom condition - use it please:

```javascript
useBrowserSupport({
  onSupported: doSomeIfSupported,
  onUnsupported: doSomeIfUnsupported,

  onCheck: isSupported => {
    // if non - IOS - just returning supported flag determined by the hook
    if (!isIOS) {
      return isSupported
    }

    return your custom condition for IOS
  }  
})
```